pkg/types/validation/installconfig_test.go

```diff
@@ -3022,11 +3042,24 @@ type installConfigBuilder struct {
 }
 
 func installConfig() *installConfigBuilder {
+	bmPlatform := validBareMetalPlatform()
+	bmPlatform.Hosts = nil
 	return &installConfigBuilder{
-		InstallConfig: types.InstallConfig{},
+		InstallConfig: types.InstallConfig{Platform: types.Platform{BareMetal: bmPlatform}},
 	}
 }
 
+func (icb *installConfigBuilder) PlatformAWS() *installConfigBuilder {
+	icb.InstallConfig.Platform.BareMetal = nil
```

Comment:
nit: we can remove this since we replace the entire `platform` field in the next line?